lib/dl_file_secure_reader_lib/dl_file_secure_reader_lib/resources/reader.py

```diff
@@ -18,6 +22,34 @@
 def parse_excel_data(data: BinaryIO, feature_excel_read_only: bool) -> list:
     result = []
 
+    # Cache wrappers for cell values to prevent excessive memory usage when
+    #  excel file is made of duplicate values.
+    cache_values: dict[tuple[str, Any], Any] = {}
+
+    def process_cell(cell: openpyxl.cell.cell.Cell) -> dict:
+        if cell.data_type == "d":
+            cell_value = str(cell.value)
+        else:
+            cell_value = cell.value
+
+        if isinstance(cell.value, int):
+            cell_type = "i"
+        else:
+            cell_type = cell.data_type
+
+        cache_key = (cell_type, cell_value)
+
+        if cache_key in cache_values:
+            cache_value = cache_values[cache_key]
+        else:
+            cache_value = {
```

Comment:
let's make it mutable dict to prevent possible mutations if we are using same value for multiple cells